bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -199,6 +199,13 @@ private void handlePurgeRequest(PurgeRequestTuple request, final AbstractLengthP
     }
   }
 
+  public void purge(long windowId)
+  {
+    for (DataList dataList: publisherBuffers.values()) {
```

Comment:
Purge inside the *Block* is not thread safe, but the Purge inside the *DataList* calls the Block purge using Synchronized block. So it should be fine.